service/src/test/java/com/epam/rft/atsy/service/impl/CandidateServiceImplTest.java

```diff
@@ -280,6 +297,137 @@ public void getAllCandidateShouldCallRepositoryWithDescendingSortWhenPassingDesc
+        CandidateFilterRequest.builder().pageNumber(null).build();
+    // When
+    candidateService.getCandidatesByFilterRequest(candidateFilterRequest);
+    // Then
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void getCandidatesByFilterRequestShouldThrowIllegalARgumentExceptionWhenSortNameIsNotNullButSortOrderIsNull() {
+    // Given
+    CandidateFilterRequest
+        candidateFilterRequest =
+        CandidateFilterRequest.builder().sortName(SORT_NAME_NAME).sortOrder(null).build();
+    // When
+    candidateService.getCandidatesByFilterRequest(candidateFilterRequest);
+    // Then
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void getCandidatesByFilterRequestShouldThrowIllegalARgumentExceptionWhenSortOrderIsNotNullButSortNameIsNull() {
+    // Given
+    CandidateFilterRequest
+        candidateFilterRequest =
+        CandidateFilterRequest.builder().sortOrder(SORT_ORDER_ASC).sortName(null).build();
+    // When
+    candidateService.getCandidatesByFilterRequest(candidateFilterRequest);
+    // Then
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void getCandidatesByFilterRequestShouldThrowIllegalARgumentExceptionWhenSortOrderIsInvalid() {
+    // Given
+    CandidateFilterRequest
+        candidateFilterRequest =
+        CandidateFilterRequest.builder().sortName(SORT_NAME_NAME).sortOrder(INVALID_SORT_ORDER)
+            .build();
+    // When
+    candidateService.getCandidatesByFilterRequest(candidateFilterRequest);
+    // Then
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void getCandidatesByFilterRequestShouldThrowIllegalARgumentExceptionWhenSortNameIsInvalid() {
+    // Given
+    CandidateFilterRequest
+        candidateFilterRequest =
+        CandidateFilterRequest.builder().sortName(INVALID_SORT_NAME).sortOrder(SORT_ORDER_ASC)
+            .build();
+    // When
+    candidateService.getCandidatesByFilterRequest(candidateFilterRequest);
+    // Then
+  }
+
+  @Test
+  public void getCandidatesByFilterRequestShouldFindDummyCandidateDTOInAscendingOrderByNameWhenThereIsARightCandidateFilterRequest() {
+    // Given
+
+    CandidateFilterRequest
+        candidateFilterRequest =
+        CandidateFilterRequest.builder().pageNumber(DEFAULT_PAGE_NUMBER).pageSize(DEFAULT_PAGE_SIZE)
+            .sortName(SORT_NAME_NAME).sortOrder(SORT_ORDER_ASC).candidateName(NAME).build();
+
+    PageRequest
+        pageRequest =
+        new PageRequest(candidateFilterRequest.getPageNumber(),
+            candidateFilterRequest.getPageSize(),
+            Sort.Direction.fromString(candidateFilterRequest.getSortOrder()),
+            candidateFilterRequest.getSortName());
+
+    String name = candidateFilterRequest.getCandidateName();
+    String email = candidateFilterRequest.getCandidateEmail();
+    String phone = candidateFilterRequest.getCandidatePhone();
+    String positions = candidateFilterRequest.getCandiadtePositions();
```

Comment:
Minor improvement suggestion: these local variables (pageRequest, name, email, phone, positions) can be final.